packages/core/src/internal/effectTypes.js

```diff
@@ -1,3 +1,4 @@
+export const ROOT = 'ROOT'
```

Comment:
i dont think I quite like this change - seems like `ROOT` is different from other effects because its purely internal

could u describe what do u think about it? what are advantages of having it in your opinion? (remember that I can always be persuaded! 😉 )

Reply:
From the perspective of the people who uses redux-saga, ROOT is a different and should be kept internal only; From the perspective of saga-monitor, ROOT is of no difference compared to ALL/RACE/CALL.

IMO, ROOT is a valid effect type, but it does not need to be exported because users never yield a root effect directly. I'll remove it from *effectTypes.js*.

Reply:
if we decide to keep it then it should stay in effectTypes, it shouldnt be exported from effects (as in `import {root} from 'redux-saga/effects'`

there is also a perspective of redux-saga itself - this effect is not interpreted like other effects, its executed eagerly on the startup

Reply:
Actually users can not do `import {root} from 'redux-saga/effects'` as `root` is not re-exported in core/src/effects.js.

I'm a little bit of confused about what to do now... Should we add ROOT as a normal effect like ALL/RACE? Or should we remove ROOT from effectTypes, and keep `root` as internal? Or do you have any nicer ideas?

Reply:
im quite sure it shouldnt be exported, im still unsure if it should be an effect at all 🤔 this only seems to satisfy monitor needs, from core's perspective it still seems to be weird imho - @restrry  wdyt?

Reply:
OTOH it is super minor thing, and was classified as an effect already when passing info about it to the monitor

Reply:
imho that's not an effect per se, as it  has other lifecycle for end user (import - call - processing - return a result) and more over, users don't know about it, as it internal detail.

Reply:
it could be kept as purely internal effect - not user facing, but I think I like `rootSagaStarted` hook proposal by @shinima better, it's more explicit and less confusing about what effects are

Reply:
+1